index.js

```diff
@@ -889,17 +740,52 @@ app.post(
   },
 )
 
+app.post('/api/heartbeat', Validator.body(['coreid', 'event', 'data']).exists(), async (request, response) => {
+  try {
+    const validationErrors = Validator.validationResult(request).formatWith(helpers.formatExpressValidationErrors)
+
+    if (validationErrors.isEmpty()) {
+      const coreId = request.body.coreid
+      const location = await db.getLocationFromParticleCoreID(coreId)
+      if (!location) {
+        const errorMessage = `Bad request to ${request.path}: no location matches the coreID ${coreId}`
+        helpers.logError(errorMessage)
+        // Must send 200 so as not to be throttled by Particle (ref: https://docs.particle.io/reference/device-cloud/webhooks/#limits)
+        response.status(200).json(errorMessage)
+      } else {
+        const message = JSON.parse(request.body.data)
+        const doorStatus = message.door_status
+        const doorTime = message.door_time
+        const insTime = message.ins_time
```

Comment:
Should this be an error if any of these values are empty? If so, maybe they should have a Validator.